src/analyze/analyze-verify.c

```diff
@@ -218,21 +231,39 @@ static int verify_unit(Unit *u, bool check_man) {
         return r;
 }
 
-int verify_units(char **filenames, UnitFileScope scope, bool check_man, bool run_generators) {
+static void restore_log_syntax_callback(void *userdata) {
+        if (userdata)
+                set_log_syntax_callback(/* cb= */ log_syntax_callback, /* userdata= */ userdata);
+}
```

Comment:
Hmm, this doesn't do what it's supposed to do. This is called with a pointer to local variable (`&temp`). You can't restore the callback with just one pointer. But I don't think we need to try to "restore" the callback, we wouldn't make use of this functionality with the current code. So just unset it:

```C
static void clear_log_syntax_callback(void *dummy) {
          set_log_syntax_callback(NULL, NULL);
}
```